src/OrchardCore.Modules/OrchardCore.Layers/Views/Admin/Index.cshtml

```diff
@@ -66,11 +67,9 @@
 
                             @if (Model.Layers.Count > 0)
                             {
-                                var widgetContentTypes = (await ContentDefinitionManager.ListTypeDefinitionsAsync())
-                                .Where(t => t.StereotypeEquals("Widget"))
-                                .ToList();
+                                var widgetContentTypes = await ContentDefinitionManager.ListWidgetTypeDefinitionsAsync();
 
-                                if (widgetContentTypes.Count > 0)
+                                if (widgetContentTypes.Any())
```

Comment:
This will do a double-enumerations. `ToList`, or rather, `ToArray` and using `Count` is better.

Reply:
no need to materialize the enumerable just to get the total could. Enumerable is this case is better because we don't need to materialize. https://github.com/OrchardCMS/OrchardCore/discussions/15240#discussioncomment-8348547

Reply:
We foreach over it a few lines below.

Reply:
yes using the Enumerable without needing to materialize.

Reply:
But we still enumerate it twice. That's my point. This is also something that ReSharper and analyzers tell about. The benchmark you linked is comparing single enumerations.

Reply:
yes that's okay to enumerate multiple times because the `.Any()` would evaluate the first entry and stops. then we do a full enumeration. This is better to avoid materializing. I don't know why would `ReSharper` complains here. It should not. It should only complain if you use `.Any()` with a materialized collection like list or array. If the record is materialized already, then you SHOULD use Length or Count instead of `.Any()`.

Reply:
Alright!